### PR TITLE
Diag layer refactoring spinoffs

### PR DESCRIPTION
**Pull request by andlaus**
This PR contains most (i.e., all I found) of the changes of #326 which are not or only tangentially related to refactoring the `DiagLayer` class into a class hierarchy corresponding to the ODX specification.

Andreas Lauser &lt;andreas.lauser@mercedes-benz.com&gt;, on behalf of [MBition GmbH](https://mbition.io/).
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

File: odxtools/comparamspec.py [hunk 1]
```diff
@@ -51,8 +51,7 @@ def from_et(et_element: ElementTree.Element, doc_frags: List[OdxDocFragment]) ->
 
     def _build_odxlinks(self) -> Dict[OdxLinkId, Any]:
         odxlinks: Dict[OdxLinkId, Any] = {}
-        if self.odx_id is not None:
-            odxlinks[self.odx_id] = self
+        odxlinks[self.odx_id] = self
```
Comment (kayoub5) [inline odxtools/comparamspec.py:54]: Even in ODX 2.0 ?

File: odxtools/comparamspec.py [hunk 2]
```diff
@@ -26,7 +26,7 @@ class ComparamSpec(IdentifiableElement):
     def from_et(et_element: ElementTree.Element, doc_frags: List[OdxDocFragment]) -> "ComparamSpec":
 
         short_name = odxrequire(et_element.findtext("SHORT-NAME"))
-        doc_frags = [OdxDocFragment(short_name, str(et_element.tag))]
+        doc_frags = [OdxDocFragment(short_name, "COMPARAM-SPEC")]
```
Comment (kayoub5) [inline odxtools/comparamspec.py:29]: I think this will break in ODX 2.0

File: odxtools/comparamspec.py (hunk 1 above)
Comment (andlaus) [reply odxtools/comparamspec.py:54]: I just checked: in ODX 2.0, COMPARAM-SPEC is derived from ODX-CATEGORY for which the ID attribute is mandatory.

tl;dr: yes

File: odxtools/comparamspec.py (hunk 2 above)
Comment (andlaus) [reply odxtools/comparamspec.py]: ok, I removed the patch...

File: odxtools/servicebinner.py [hunk 3]
```diff
@@ -91,7 +91,7 @@ def __repr__(self) -> str:
         """
         result = StringIO()
         result.write("[ ")
-        result.write(", ".join([f"0x{x}" for x in self._service_groups if x is not None]))
+        result.write(", ".join([f"0x{x:x}" for x in self._service_groups if x is not None]))
```
[the remark this answers is not in the document]
Comment (andlaus) [reply odxtools/servicebinner.py:94]: the standard way for such things is to use format specifiers in f-strings: e.g. printing a floating point value with two decimal places accuracy and filled up with zeros would be `[float_val:.03f}`. there are already a few instances of this in the code base, so I think we should use the same mechanism everywhere...

**Review by kayoub5 — APPROVED** [review]
